react/src/components/forms/Input/index.js

```diff
@@ -16,18 +17,17 @@ const Input = (props) => {
     'ma__input-group--inline': inline,
     [classes && classes.length > 0 && classes.join(' ')]: true
   });
-  const inputLabelClasses = classNames({
-    ma__label: true,
-    'ma__label--hidden': (labelText && hiddenLabel),
-    'ma__label--required': (labelText && required),
-    'ma__label--optional': (labelText && !required),
-    'ma__label--disabled': (labelText && disabled)
-  });
+  const conditionText = required ? null : 'optional';
```

Comment:
Technically this should be set to empty string instead of null. The Label component should check to see if the length of condtionText is greater than zero, since the propType is set to be a string and not null.